docs/src/unstructured_quad_mesh/unstructured_quad_mesh.md

```diff
@@ -0,0 +1,179 @@
+# Unstructured quadrilateral mesh
+
+Herein we describe the conventions taken in the implementation for two-dimensional unstructured
+quadrilateral meshes. Principally, this relates to how a file with the extension `.mesh` encodes
+information about the numbering and orientation of elements in an unstructured quadrilateral mesh
+with possibly curved boundaries.
+
+We use the following unstructured mesh with three elements for this discussion:
+
+![example-mesh](example_mesh.png)
```

Comment:
Why shouldn't we handle it this way right now? @andrewwinters5000 is working in his fork and we will squash-merge this PR, so I guess this will be okay for us? Hence, it would be good if you could post the picture here, grab the link, and use that in the docs, @andrewwinters5000.

Reply:
Ah, right! I forgot about squash-merging! Yes, in this case, please replace the uploaded image by a link to a posted picture (you can just drag and drop it in the general comment section of this PR - this will yield the link; you should still post the comment to ensure that GH won't clean up the file at some point because it is not referenced)

Reply:
So the link to the figure if I post it here never expires? This seems like a real dodgy workaround...

Reply:
I don't see how it can expire unless GitHub decides to sneakily edit our posts and changes the URL there. It is a dodgy workaround, I agree, but at the moment there are so many high(er) priority items on my Trixi TODO list that I doubt I'll get to it anytime soon.